Scripts/RemediationScripts/Remediate-PTAControls.ps1

```diff
@@ -0,0 +1,915 @@
+    Write-Host "Checking if the required modules are present..."
+    # List of available modules
+    $availableModules = $(Get-Module -ListAvailable $requiredModules -ErrorAction Stop)
+
+    # Check if the required modules are installed.
+    if ($availableModules.Name -contains "Az.Resources")
+    {
+        $module = get-module "Az.Resources"
+        if($module.Version -ge "5.0.0")
+        {
+            Write-Host " Az.Resources module is present ..." -ForegroundColor $([Constants]::MessageType.Info)        
+        }
+        else
+        {
+            Write-Host "Installing $($_) module with required version(5.0.0)..." -ForegroundColor $([Constants]::MessageType.Info)
+            Install-Module -Name "Az.Resources" -MinimumVersion 5.0.0 -RequiredVersion 5.0.0 -Scope CurrentUser -Repository 'PSGallery'  -Force -ErrorAction Stop   
+        }
+    }
+    else
+    {
+        Write-Host "Installing $($_) module...." -ForegroundColor $([Constants]::MessageType.Update)
+        Install-Module -Name "Az.Resources" -MinimumVersion 5.0.0 -RequiredVersion 5.0.0 -Scope CurrentUser -Repository 'PSGallery' -Force -ErrorAction Stop
+    }
+
+
+    if($availableModules.Name -contains "AZ.Accounts")
+    {
+        $module = get-module "Az.Accounts"
+        if($module.Version -ge "2.5.4")
+        {
+            Write-Host "Az.Accounts module is present ... " -ForegroundColor $([Constants]::MessageType.Info)
+        }
+        else
+        {
+             Write-Host "Installing Az.Accounts module with required version(2.5.4)..." -ForegroundColor $([Constants]::MessageType.Info)
+             Install-Module -Name "Az.Accounts" -MinimumVersion 2.5.4 -RequiredVersion 2.5.4 -Scope CurrentUser -Repository 'PSGallery' -Force -ErrorAction Stop        
+        }    
+    }
+    else
+    {
+        Write-Host "Installing Az.Accounts module...." -ForegroundColor $([Constants]::MessageType.Update)
+        Install-Module -Name "Az.Accounts" -MinimumVersion 2.5.4 -RequiredVersion 2.5.4 -Scope CurrentUser -Repository 'PSGallery'  -Force -ErrorAction Stop
+    }
+
+    if($availableModules.Name -contains "AzureAD")
+    {
+        Write-Host "AzureAD module is present ..." -ForegroundColor $([Constants]::MessageType.Info)
+    }
+    else
+    {
+        Write-Host "Installing AzureAD module...." -ForegroundColor $([Constants]::MessageType.Update)
+        Install-Module -Name AzureAD -Scope CurrentUser -Repository "PSGallery" -Force -ErrorAction Stop
+    }
+}
+
+
+function Create-PIMForPermanentRoleAssignments
+{
+    <#
+        .SYNOPSIS
+        Remediates 'Azure_Subscription_AuthZ_Dont_Grant_Persistent_Access , Azure_Subscription_AuthZ_Dont_Grant_Persistent_Access_RG' Control.
+
+        .DESCRIPTION
+        Remediates 'Azure_Subscription_AuthZ_Dont_Grant_Persistent_Access , Azure_Subscription_AuthZ_Dont_Grant_Persistent_Access_RG' Control.
```

Comment:
Should this also include the SC-ALT Control ID?
To be modified to - "PIM non SCALT role assignment(s) **and** remove"
Please fix `-resourcegroup` as per the conventions followed with others.